client/network/src/behaviour.rs

```diff
@@ -74,22 +80,40 @@ pub enum BehaviourOut<B: BlockT> {
 	RandomKademliaStarted(ProtocolId),
 
 	/// We have received a request from a peer and answered it.
-	AnsweredRequest {
+	///
+	/// This event is generated for statistics purposes.
+	InboundRequest {
 		/// Peer which sent us a request.
 		peer: PeerId,
 		/// Protocol name of the request.
-		protocol: Vec<u8>,
-		/// Time it took to build the response.
-		build_time: Duration,
+		protocol: Cow<'static, str>,
```

Comment:
While libp2p allows bytes, I went for restricting this to strings in Substrate because:

- In practice they're always strings anyway.
- These protocol names are ported to Prometheus, and Prometheus expects strings (which is the reason for `maybe_utf8_bytes_to_string`).